### PR TITLE
Add OnlyFetchMvbox option

### PR DESCRIPTION
**Pull request by Hocuri**
Depends on ~~https://github.com/deltachat/deltachat-core-rust/pull/3014~~ https://github.com/deltachat/deltachat-core-rust/pull/3028

### CODE REVIEW

**Comment by r10s** [comment]
maybe, for now, move that to "Experimental" options, so that we do not again raise wrong expectations again, if that stands in the way of something.

(but that is no blocker and can also be done in a subsequent pr, maybe together with the other pretty minor thing "Watch Send Folder")

**Comment by Hocuri** [comment]
I think that it should be close to "Watch Sent Folder" and "Automatic Moves to DeltaChat Folder", because toggling it also toggles these two. If we want to make clear that it's experimental, maybe write "(Experimental)" before the summary?

File: res/values/strings.xml [hunk 1]
```diff
@@ -607,6 +607,8 @@
     <string name="pref_send_copy_to_self">Send Copy to Self</string>
     <string name="pref_auto_folder_moves">Automatic Moves to DeltaChat Folder</string>
     <string name="pref_auto_folder_moves_explain">Chat conversations are moved to avoid cluttering the Inbox</string>
+    <string name="pref_only_fetch_mvbox_title">Only fetch from the DeltaChat folder</string>
+    <string name="pref_only_fetch_mvbox_explain">This will ignore all other folders. Regardless of this setting, the Spam folder is sometimes checked for replies to known messages.</string>
```
Comment (Hocuri) [inline res/values/strings.xml:611]: I'm wondering if we should change this to

```suggestion
    <string name="pref_only_fetch_mvbox_explain">…and ignore all other folders. Regardless of this setting, the Spam folder is sometimes checked for replies to known messages.</string>
```

Comment (r10s) [reply res/values/strings.xml:611]: i do not like the ellipsis-form - we do not use that somewhere else, it is overall not really used and looks a bit weird. also may make translations a bit harder as a sentence is split.

as the plain "Ignore all other folders." is even shorter, i'd vote for that :)

**Review by adbenitez — APPROVED** [review]
looks good, but didn't test

File: res/values/strings.xml [hunk 2]
```diff
@@ -607,6 +607,8 @@
     <string name="pref_send_copy_to_self">Send Copy to Self</string>
     <string name="pref_auto_folder_moves">Automatic Moves to DeltaChat Folder</string>
     <string name="pref_auto_folder_moves_explain">Chat conversations are moved to avoid cluttering the Inbox</string>
+    <string name="pref_only_fetch_mvbox_title">Only fetch from the DeltaChat folder</string>
```
Comment (r10s) [inline res/values/strings.xml:610]:
```suggestion
    <string name="pref_only_fetch_mvbox_title">Only Fetch from DeltaChat Folder</string>
```

we "headline capitalize" options meanwhile, that means everythings beside to/and/or/from/etc. becomes uppercase

File: res/values/strings.xml [hunk 3]
```diff
@@ -607,6 +607,8 @@
     <string name="pref_send_copy_to_self">Send Copy to Self</string>
     <string name="pref_auto_folder_moves">Automatic Moves to DeltaChat Folder</string>
     <string name="pref_auto_folder_moves_explain">Chat conversations are moved to avoid cluttering the Inbox</string>
+    <string name="pref_only_fetch_mvbox_title">Only Fetch from DeltaChat Folder</string>
+    <string name="pref_only_fetch_mvbox_explain">Ignore all other folders. Regardless of this setting, the Spam folder is sometimes checked for replies to known messages.</string>
```
Comment (Hocuri) [inline res/values/strings.xml:611]: @r10s you wrote that the option description should mention that that requires the help of the server - like this?

```suggestion
    <string name="pref_only_fetch_mvbox_explain">This only works if your server somehow moves chat messages to the DeltaChat folder.</string>
```

File: src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java [hunk 4]
```diff
@@ -76,16 +78,28 @@ public void onCreate(Bundle paramBundle) {
     bccSelfCheckbox.setOnPreferenceChangeListener((preference, newValue) -> {
       boolean enabled = (Boolean) newValue;
       dcContext.setConfigInt(CONFIG_BCC_SELF, enabled? 1 : 0);
+      DcHelper.getAccounts(getContext()).startIo();
```
Comment (r10s) [inline src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java:81]: why is io not stopped before here?

File: src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java [hunk 5]
```diff
@@ -76,16 +78,28 @@ public void onCreate(Bundle paramBundle) {
...
       return true;
     });
 
+    onlyFetchMvboxCheckbox = this.findPreference("pref_only_fetch_mvbox");
+    onlyFetchMvboxCheckbox.setOnPreferenceChangeListener(((preference, newValue) -> {
+      boolean enabled = (Boolean) newValue;
+      DcHelper.getAccounts(getContext()).stopIo();
+      dcContext.setConfigInt(CONFIG_ONLY_FETCH_MVBOX, enabled? 1 : 0);
+      DcHelper.getAccounts(getContext()).startIo();
```
Comment (r10s) [inline src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java:99]: we should probably add the line `changes require restarting IO by calling dc_stop_io() and then dc_start_io().` to deltachat.h to make the requirement clear.

File: src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java (hunk 4 above)
Comment (Hocuri) [reply src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java:81]: Doesn't seem needed, idk why I added it

**Review by r10s — APPROVED** [review]
some minor comments ...

File: src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java (hunk 4 above)
Comment (r10s) [reply src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java]: i just had a rough look at core, it seems not to be needed.

File: src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java (hunk 5 above)
Comment (Hocuri) [reply src/org/thoughtcrime/securesms/preferences/AdvancedPreferenceFragment.java:98]: I'll do that

**Pull request merged**